include/extractor/turn_instructions.hpp

```diff
@@ -31,37 +36,98 @@ enum class TurnInstruction : unsigned char
     AccessRestrictionPenalty = 129
 };
 
-// Translates between angles and their human-friendly directional representation
-inline TurnInstruction getTurnDirection(const double angle)
+inline TurnInstruction shiftTurnToLeft(TurnInstruction turn)
 {
-    if (angle >= 23 && angle < 67)
-    {
-        return TurnInstruction::TurnSharpRight;
-    }
-    if (angle >= 67 && angle < 113)
+    switch (turn)
     {
+    case TurnInstruction::NoTurn:
+        return TurnInstruction::TurnSlightLeft;
+    case TurnInstruction::GoStraight:
+        return TurnInstruction::TurnSlightLeft;
+    case TurnInstruction::TurnSlightRight:
+        return TurnInstruction::GoStraight;
+    case TurnInstruction::TurnRight:
+        return TurnInstruction::TurnSlightRight;
+    case TurnInstruction::TurnSharpRight:
         return TurnInstruction::TurnRight;
+    case TurnInstruction::TurnLeft:
+        return TurnInstruction::TurnSharpLeft;
+    case TurnInstruction::TurnSlightLeft:
+        return TurnInstruction::TurnLeft;
+    default:
+        return turn;
     }
-    if (angle >= 113 && angle < 158)
+}
+
+inline TurnInstruction shiftTurnToRight(TurnInstruction turn)
+{
+    switch (turn)
     {
+    case TurnInstruction::NoTurn:
         return TurnInstruction::TurnSlightRight;
-    }
-    if (angle >= 158 && angle < 202)
-    {
+    case TurnInstruction::GoStraight:
+        return TurnInstruction::TurnSlightRight;
+    case TurnInstruction::TurnSlightRight:
+        return TurnInstruction::TurnRight;
+    case TurnInstruction::TurnRight:
+        return TurnInstruction::TurnSharpRight;
+    case TurnInstruction::TurnSharpLeft:
+        return TurnInstruction::TurnLeft;
+    case TurnInstruction::TurnLeft:
+        return TurnInstruction::TurnSlightLeft;
+    case TurnInstruction::TurnSlightLeft:
         return TurnInstruction::GoStraight;
+    default:
+        return turn;
     }
-    if (angle >= 202 && angle < 248)
-    {
+}
+
+inline double angularDeviation( const double angle, const double from ){
+  const double deviation = fabs( angle - from );
```

Comment:
C has no function overloading, C++ has. There should be no need for `fabs` here, `std::abs` should suffice.